ide/docker.editor/src/org/netbeans/modules/docker/editor/DockerfileDataObject.java

```diff
@@ -0,0 +1,134 @@
+            id = @ActionID(category = "System", id = "org.openide.actions.OpenAction"),
+            position = 100,
+            separatorAfter = 300
+    ),
+    @ActionReference(
+            path = "Loaders/text/x-dockerfile/Actions",
+            id = @ActionID(category = "Edit", id = "org.openide.actions.CutAction"),
+            position = 400
+    ),
+    @ActionReference(
+            path = "Loaders/text/x-dockerfile/Actions",
+            id = @ActionID(category = "Edit", id = "org.openide.actions.CopyAction"),
+            position = 500,
+            separatorAfter = 600
+    ),
+    @ActionReference(
+            path = "Loaders/text/x-dockerfile/Actions",
+            id = @ActionID(category = "Edit", id = "org.openide.actions.DeleteAction"),
+            position = 700
+    ),
+    @ActionReference(
+            path = "Loaders/text/x-dockerfile/Actions",
+            id = @ActionID(category = "System", id = "org.openide.actions.RenameAction"),
+            position = 800,
+            separatorAfter = 900
+    ),
+    @ActionReference(
+            path = "Loaders/text/x-dockerfile/Actions",
+            id = @ActionID(category = "System", id = "org.openide.actions.SaveAsTemplateAction"),
+            position = 1000,
+            separatorAfter = 1100
+    ),
+    @ActionReference(
+            path = "Loaders/text/x-dockerfile/Actions",
+            id = @ActionID(category = "System", id = "org.openide.actions.FileSystemAction"),
+            position = 1200,
+            separatorAfter = 1300
+    ),
+    @ActionReference(
+            path = "Loaders/text/x-dockerfile/Actions",
+            id = @ActionID(category = "System", id = "org.openide.actions.ToolsAction"),
+            position = 1400
+    ),
+    @ActionReference(
+            path = "Loaders/text/x-dockerfile/Actions",
+            id = @ActionID(category = "System", id = "org.openide.actions.PropertiesAction"),
+            position = 1500
+    )
+})
+
+
+@DataObject.Registration(
+        mimeType = MIME_TYPE,
+        iconBase = "org/netbeans/modules/docker/editor/resources/docker_file.png",
+        displayName = "#Dockerfile",
+        position = 303
+)
+public final class DockerfileDataObject extends MultiDataObject {
+
+    public DockerfileDataObject(FileObject pf, MultiFileLoader loader) throws DataObjectExistsException, IOException {
+        super(pf, loader);
+        registerEditor(MIME_TYPE, true);
+    }
+    @Override
+    protected Node createNodeDelegate() {
+        return new DataNode(this, Children.LEAF, getLookup());
+    }
+
+    @Override
+    protected int associateLookup() {
+        return 1;
+    }
+
+    @NbBundle.Messages("Source=&Source")
+    @MultiViewElement.Registration(
+            displayName = "#Source",
+            iconBase = "org/netbeans/modules/docker/editor/resources/docker_file.png",
+            persistenceType = 1,
+            mimeType = MIME_TYPE,
+            preferredID = "neon.source",
```

Comment:
neon? I would use "docker.source".

Reply:
This whole section is a copy+paste from Yaml support. I think I have to revisit those as well.